mux.go

```diff
@@ -344,6 +350,10 @@ type RouteMatch struct {
 	Route   *Route
 	Handler http.Handler
 	Vars    map[string]string
+
+	// MethodMismatch flag is set to true if a route is matched but there
+	// is a mismatch in the request method and route method
+	MethodMismatch bool
```

Comment:
I wonder if it would be better to have a `MatchError error` field instead. Then there could be a `ErrMethodMismatch` value used there in the case of methodMatcher, and then it could be extended with other error types in the future.